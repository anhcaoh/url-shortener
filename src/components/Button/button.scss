@import "Scss/__typography.scss";
button:disabled{
    cursor: not-allowed;
    opacity: 0.5;
}
.button{
    @include font-style();
    padding: 0.25rem 0.5rem;
    cursor: pointer;
    outline: none;
    border-radius: $radius;
    border: 1px solid #eee;
}
.button.primary{
    background-color: #1463ff;
    color: #fefefe;
    border-color: #1463ff;
    // background: #457fca;  /* fallback for old browsers */
    // background: -webkit-linear-gradient(to bottom, #5691c8, #457fca);  /* Chrome 10-25, Safari 5.1-6 */
    // background: linear-gradient(to bottom, #5691c8, #457fca); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
    // color: #f1f1f1;
    // border-color: #548fc8;
}
.button.stord{
    box-shadow: 6px 6px 0 #010e18;
    transition: .4s cubic-bezier(.2,.8,.2,1);
}