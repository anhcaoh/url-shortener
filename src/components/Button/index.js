import React from "react";
import "./button.scss";
const Button = (props) => {
    return (<button {...props} 
    type={props.type || "button"}
    className={["button", props.className].join(" ").trim()} />);
};
export default Button;