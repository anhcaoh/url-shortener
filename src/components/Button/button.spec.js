import React from "react";
import renderer from "react-test-renderer";
import Button from "./index";
import { shallow } from "enzyme";

describe("Button", () => {
  it("should render snapshot", () => {
    const component = renderer.create(<Button>Button</Button>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should handle click event 2 times", () => {
    const mockFunc = jest.fn();
    const wrapper = shallow((<Button onClick={mockFunc}>Button</Button>));
    wrapper.find("button").simulate("click");
    wrapper.find("button").simulate("click");
    expect(mockFunc).toHaveBeenCalledTimes(2);
});

});