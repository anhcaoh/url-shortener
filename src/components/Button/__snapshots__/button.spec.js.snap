// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render snapshot 1`] = `
<button
  className="button"
  type="button"
>
  Button
</button>
`;
